### PR TITLE
Update re: adding SSH key to the ssh-agent for Mac users (Issue #23159)

### LINKED ISSUE #23159: SSH doesn't use passphrase stored in keychain on MacOS Monterey, after following directions
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent

### [Adding your SSH key to the ssh-agent](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#adding-your-ssh-key-to-the-ssh-agent)
After following the directions given in this section, to store the passphrase in keychain, I continued to be prompted for the passphrase on every `git push`. I am using MacOS 12.6.2.

After reading some forums on related issues on recent MacOS versions, I found that copying the command given at step 3(`ssh-add --apple-use-keychain ~/.ssh/id_{% ifversion ghae %}rsa{% else %}ed25519{% endif %}`) into `~/.zshrc` gave correct behaviour.

Use case: Mac users encountering problems with SSH connections using a stored passphrase.

Expected outcome: a note to suggest a fix to this problem, if encountered.

I suggest adding to the note at step 3 that this extra step may be required, e.g.:

> If you continue to be prompted for your passphrase, you may need to add the above command to your `~/.zshrc` file (or `.bashrc` for bash).

### Additional information

`~/.ssh/config` is as per the instructions given, with no other contents.

### PR DESCRIPTION
**Pull request by oC-n**
Added a note to step 3 of **Adding your SSH key to the ssh-agent** for Mac users, for whom the system may continue to ask for SSH passphrase on every `git push`.

### Why:
Following the directions given under **Adding your SSH key to the ssh-agent** can result in Mac systems continuing to ask for SSH passphrase on every `git push`. The suggested edit resolves this problem, if imperfectly.

Closes: https://github.com/github/docs/issues/23159

### What's being changed (if available, include any code snippets, screenshots, or gifs):
Adding an extra line to the note for Mac users on Step 3 of **Adding your SSH key to the ssh-agent**, instructing that `ssh-add --apple-use-keychain ~/.ssh/id_ed25519` can be added to `.zshrc` or `.zshrc` to resolve the problem of being repeatedly asked for SSH passphrase.

### Check off the following:

- [x] I have reviewed my changes in staging (look for the "Automatically generated comment" and click the links in the "Preview" column to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

**Comment by cmwilson21** [comment]
@oC-n Thanks so much for opening a PR! I'll get this triaged for review :zap:

File: content/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent.md [hunk 1]
```diff
@@ -156,6 +156,8 @@ Before adding a new SSH key to the ssh-agent to manage your keys, you should hav
    The `--apple-use-keychain` option is in Apple's standard version of `ssh-add`. In MacOS versions prior to Monterey (12.0), the `--apple-use-keychain` and `--apple-load-keychain` flags used the syntax `-K` and `-A`, respectively.
 
   If you don't have Apple's standard version of `ssh-add` installed, you may receive an error. For more information, see "[Error: ssh-add: illegal option -- K](/articles/error-ssh-add-illegal-option-k)."
+  
+  If you continue to be prompted for your passphrase, you may need to add the above command to your ~/.zshrc file (or .bashrc for bash).
```
Comment (am-stead) [inline content/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent.md:160]:
```suggestion
  If you continue to be prompted for your passphrase, you may need to add the above command to your `~/.zshrc` file (or your `~/.bashrc` file for bash).
```

**Review by am-stead — COMMENTED** [review]
Thanks for submitting this PR!✨ 
I've made a small suggestion so that it aligns with our style guide!

**Review by am-stead — DISMISSED** [review]
I've applied the stylistic edit and I have approved the PR for merging ✨

File: content/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent.md [hunk 2]
```diff
@@ -155,7 +155,9 @@ Before adding a new SSH key to the ssh-agent to manage your keys, you should hav
 
    The `--apple-use-keychain` option is in Apple's standard version of `ssh-add`. In MacOS versions prior to Monterey (12.0), the `--apple-use-keychain` and `--apple-load-keychain` flags used the syntax `-K` and `-A`, respectively.
 
-   If you don't have Apple's standard version of `ssh-add` installed, you may receive an error. For more information, see "[Error: ssh-add: illegal option -- K](/articles/error-ssh-add-illegal-option-k)."
+  If you don't have Apple's standard version of `ssh-add` installed, you may receive an error. For more information, see "[Error: ssh-add: illegal option -- K](/articles/error-ssh-add-illegal-option-k)."
+  
+  If you continue to be prompted for your passphrase, you may need to add the above command to your `~/.zshrc` file (or your `~/.bashrc` file for bash).
```
Comment (am-stead) [inline content/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent.md:160]:
```suggestion
  If you continue to be prompted for your passphrase, you may need to add the command to your `~/.zshrc` file (or your `~/.bashrc` file for bash).
```

**Review by am-stead — APPROVED** [review]

**Pull request merged**